Fix author visibility issue in the editor.

rohitmathur-7 opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Fixed the Author Visibility issue by displaying the alphabetically first author's name when the post/page is created by a super admin that does not have their user registered in the sub-site.
Fixes the issue: https://github.com/WordPress/gutenberg/issues/62574

## Why?
There is a UI issue if we are not able to see any author's name. It fixes that.

## How?
When a super admin adds a new post/page and its user is not registered on the sub-site, then instead of showing nothing for the author I have added the name of the alphabetically first author's name. 

## Testing Instructions
- Login to your WordPress site using a Super admin user.
- Go on a sub-site where the Super admin user you are using does not have its user registered.
- Open a new post or page.
- Check the author in the right panel of the editor.

## Screenshots or screencast <!-- if applicable -->
<img width="294" alt="author-fix" src="https://github.com/WordPress/gutenberg/assets/62138457/2665f7ab-a440-453d-bb36-16b7d21700cb">
<img width="1470" alt="Screenshot 2024-06-14 at 4 16 26 PM" src="https://github.com/WordPress/gutenberg/assets/62138457/d9692a95-0bee-4695-b700-00783078844c">

rohitmathur-7 committed:
Fixed the author visibility issue in the posts editor by adding the name of the alphabetically first author in the Author's name when that field is empty

t-hamano reviewed (commented):
Thanks for the PR!

> then instead of showing nothing for the author I have added the name of the alphabetically first author's name.

I think this might be confusing because it shows a user who isn't the actual user who created the post (super admin).

I'd like to suggest an alternative approach. See [the comment](https://github.com/WordPress/gutenberg/issues/62574#issuecomment-2168010414) here for more details.

rohitmathur-7 commented:
> Thanks for the PR!
> 
> > then instead of showing nothing for the author I have added the name of the alphabetically first author's name.
> 
> I think this might be confusing because it shows a user who isn't the actual user who created the post (super admin).
> 
> I'd like to suggest an alternative approach. See [the comment](https://github.com/WordPress/gutenberg/issues/62574#issuecomment-2168010414) here for more details.

Sorry I just saw the comment on the PR now, and I added the name of the alphabetically first author's name because in the previous versions if a super admin created a post then it showed alphabetically first author's name as the author, so to maintain consistency I did the same.

t-hamano commented:
> because in the previous versions if a super admin created a post then it showed alphabetically first author's name as the author

Oh, I didn't know that. Do you know in which version of WordPress this issue first occurred?

rohitmathur-7 commented:
> > because in the previous versions if a super admin created a post then it showed alphabetically first author's name as the author
> 
> Oh, I didn't know that. Do you know in which version of WordPress this issue first occurred?

No I don't have idea in which WordPress version this first occurred.

t-hamano commented:
Sorry for the late reply.

To fundamentally solve this problem, I think some kind of work needs to be done on the core side. See [this comment](https://github.com/WordPress/gutenberg/issues/62574#issuecomment-2285317506) for details.

Mamaduka commented:
Closing as not applicable. Displaying the first author instead of the actual post author is misleading.

As noted in discussion this requires resolution on the REST API side.
